File: CSharp/SpaceVIL/SpaceVIL/Items/Triangle.cs

namespace SpaceVIL
{
    /// <summary>
    /// Triangle is a subclass that extends from SpaceVIL.Primitive 
    /// for rendering an triangle shape.
    /// </summary>
    public class Triangle : Primitive
    {
        static int count = 0;

        /// <summary>
        /// Default Triangle constructor.
        /// </summary>
        public Triangle()
            : base(name: "Triangle_" + count)
        {
            count++;
        }

        /// <summary>
        /// Constructs an Triangle with specified rotation angle of an triangle shape.
        /// </summary>
        /// <param name="angle">Rotation angle of an triangle shape.</param>
        public Triangle(int angle) : this()
        {
            RotationAngle = angle;
        }

        /// <summary>
        /// Rotation angle in degrees of an triangle shape. 
        /// <para/> Default: 0.
        /// </summary>
        public int RotationAngle = 0;

        /// <summary>
        /// Overridden method for stretching the triangle shape relative to the current size. 
        /// Use in conjunction with GetTriangles() and SetTriangles() methods.
        /// <para/> Notice: This method is mainly for overriding only. SpaceVIL calls 
        /// this method if necessary and no need to call it manually.
        /// </summary>
        public override void MakeShape()
        {
            SetTriangles(GraphicsMathService.GetTriangle(GetWidth(), GetHeight(), 0, 0, RotationAngle));
        }
    }
}

File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Triangle.java
package com.spvessel.spacevil;

/**
 * Triangle is a subclass that extends from com.spvessel.spacevil.Primitive for
 * rendering an triangle shape.
 */
public class Triangle extends Primitive {
    private static int count = 0;
    /**
     * Rotation angle in degrees of an triangle shape.
     * <p>
     * Default: 0.
     */
    public int rotationAngle = 0;

    /**
     * Default Triangle constructor.
     */
    public Triangle() {
        setItemName("Triangle_" + count);
        count++;
    }

    /**
     * Constructs an Triangle with specified rotation angle of an triangle shape.
     * 
     * @param angle Rotation angle of an triangle shape.
     */
    public Triangle(int angle) {
        this();
        rotationAngle = angle;
    }

    /**
     * Overridden method for stretching the triangle shape relative to the current
     * size. Use in conjunction with getTriangles() and setTriangles() methods.
     * <p>
     * Notice: This method is mainly for overriding only. SpaceVIL calls this method
     * if necessary and no need to call it manually.
     */
    @Override
    public void makeShape() {
        setTriangles(GraphicsMathService.getTriangle(getWidth(), getHeight(), 0, 0, rotationAngle));
    }
}